repo_checker.py

```diff
@@ -70,6 +70,129 @@ def project_only(self, project, post_comments=False):
         state_hash = self.repository_state(repository_pairs, False)
         self.repository_check(repository_pairs, state_hash, False, bool(post_comments), arch_whitelist=arch_whitelist)
 
+    def staging(self, target, project):
+        repository = 'standard'
+        repository_pairs = repository_path_expand(self.apiurl, project, repository)
+        staging_pair = [project, repository]
+
+        result = True
+        config = Config.get(self.apiurl, target)
+        staging = config.get('staging')
+        arch_whitelist = config.get('repo_checker-arch-whitelist')
+        cycle_packages = config.get('repo_checker-allowed-in-cycles')
+        api = self.staging_api(staging)
+
+        status = api.project_status(project)
+        if not status:
+            self.logger.error('no project status for {}'.format(project))
+            return False
+
+        meta = api.load_prj_pseudometa(status['description'])
+        for req in meta['requests']:
+            comment_lines = []
+            if req['type'] == 'delete':
+                package = req['package']
+                built_binaries = set([])
+                file_infos = []
+                for fileinfo in fileinfo_ext_all(self.apiurl, target, 'standard', 'x86_64', package):
+                    built_binaries.add(fileinfo.find('name').text)
+                    file_infos.append(fileinfo)
+
+                for fileinfo in file_infos:
+                    for provides in fileinfo.findall('provides_ext'):
+                        for requiredby in provides.findall('requiredby[@name]'):
+                            if requiredby.get('name') in built_binaries:
+                                continue
+                            # extract >= and the like
+                            provide = provides.get('dep')
+                            provide = provide.split(' ')[0]
+                            self.logger.info('{} provides {} required by {}'.format(fileinfo.find('name').text, provide, requiredby.get('name')))
+                            provide_missing = True
+                            url = osc.core.makeurl(self.apiurl,
+                                          ['build', target, 'standard', 'x86_64', '_repository', requiredby.get('name') + '.rpm'],
+                                          {'view': 'fileinfo_ext'})
+                            reverse_fileinfo = ET.parse(osc.core.http_GET(url)).getroot()
+                            for require in reverse_fileinfo.findall('requires_ext'):
+                                # extract >= and the like here too
+                                dep = require.get('dep').split(' ')[0]
+                                if dep != provide:
+                                    continue
+                                for provided_by in require.findall('providedby'):
+                                    if provided_by.get('name') in built_binaries:
+                                        continue
```

Comment:
wow, 41 spaces :)

Reply:
yeah, I was trying to squeeze it all in one function :)